Sources/Examples/ExploringPerformanceExample.swift

```diff
@@ -0,0 +1,201 @@
+import Foundation
+import Logging
+
+// MARK: Example library code
+
+struct DataContainer {
+    let data: String // this would be some user `T` type
+
+    // would be W3C's `traceparent` for example
+    let traceContext: TraceContext?
+}
+
+
+// there's a few ways to specify those, this is just an example;
+//
+// See others: https://github.com/opentracing/specification/blob/master/rfc/trace_identifiers.md
+// Though this one is going to be the standardized one soon: https://w3c.github.io/trace-context/
+struct TraceContext {
+    let version: UInt8
+    let traceId: (UInt64, UInt64) // one value in 128 bits
+    let parentId: UInt64
+    let flags: UInt8
+}
+
+extension TraceContext: CustomStringConvertible {
+    public var description: String {
+        let v = String(self.version, radix: 16, uppercase: true)
+        let t1 = String(self.traceId.0, radix: 16, uppercase: true)
+        let t2 = String(self.traceId.1, radix: 16, uppercase: true)
+        let s = String(self.parentId, radix: 16, uppercase: true)
+        let f = String(self.flags, radix: 16, uppercase: true)
+
+        // rendering just an example, did not super closely follow any spec, but close enough:
+        return "\(v)-\(t1)\(t2)-\(s)-\(f)"
+    }
```

Comment:
"expensive rendering" example; It's not super super expensive, but if you imagine there being many values which have an efficient representation which then need to print themselfes constantly even if we don't actually do print them them you get a feel of how big the waste is.